src/main/kotlin/com/cognifide/gradle/aem/common/file/transfer/FileTransferManager.kt

```diff
@@ -187,6 +187,15 @@ class FileTransferManager(private val aem: AemExtension) : FileTransfer {
     override val name: String
         get() = NAME
 
+    init {
+        // override specific credentials if common specified
```

Comment:
this have precedence over specific ones; case from issue cannot be easily implemented.
i am opting for this one as of it is easily implementableas follows